### PR TITLE
Improve NestedTensor documentation

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85186

### CODE REVIEW

File: docs/source/nested.rst [hunk 1]
```diff
@@ -39,6 +55,15 @@ nested_tensor([
   tensor([3., 4., 5., 6., 7.], device='cuda:0')
 ])
 
+Autograd support is available by setting the `requires_grad` attribute. There is
+a subtlety here where the usual `requires_grad` kwarg for constructors is not supported for
+the `nested_tensor` constructor.
+
+>>> nt.requires_grad_(True)
+
+The tensors passed to the constructor can also require gradients, and autograd will
```
Comment (cpuhrsch) [inline docs/source/nested.rst:64]: I think this is an issue. We should align `torch.nested_tensor` with `torch.tensor` and `torch.as_tensor`. [`torch.tensor`](https://pytorch.org/docs/master/generated/torch.tensor.html#torch.tensor) constructs a leaf Tensor with no autograd history. As such `torch.nested_tensor` should also construct a leaf Tensor. However, the user can use `torch.as_nested_tensor` to get autograd, just like they do with [`torch.as_tensor`](https://pytorch.org/docs/master/generated/torch.as_tensor.html#torch.as_tensor).

Comment (cpuhrsch) [reply docs/source/nested.rst:64]: cc @drisspg @albanD

Comment (mikaylagawarecki) [reply docs/source/nested.rst:64]: Given the way the `torch.nested_tensor` constructor is defined (as a native_function) would it be possible to add the `requires_grad` kwarg to `torch.tensor`? Or would this be done by wrapping in a python class?

Comment (cpuhrsch) [reply docs/source/nested.rst:64]: It's possible to add kwargs to native functions, so we could add `requires_grad` if we wanted.

File: docs/source/nested.rst [hunk 2]
```diff
@@ -16,7 +16,23 @@ The only constraint on the input Tensors is that their dimension must match.
 
 This enables more efficient metadata representations and access to purpose built kernels.
 
-Construction is straightforward and involves passing a list of Tensors to the constructor.
+One application of NestedTensors is to express sequential data in various domains.
+While the conventional approach is to pad variable length sequences, NestedTensor
+enables users to bypass padding. The API for calling operations on a nested tensor is no different
+from that of a regular `torch.Tensor`, which should allow seamless integration with existing models,
+with the main difference being :ref:`construction of the inputs <construction>` of the inputs.
```
Comment (albanD) [inline docs/source/nested.rst:23]: Isn't `of the inputs` said twice here?

File: docs/source/nested.rst [hunk 3]
```diff
@@ -16,7 +16,23 @@ The only constraint on the input Tensors is that their dimension must match.
...
-Construction is straightforward and involves passing a list of Tensors to the constructor.
+One application of NestedTensors is to express sequential data in various domains.
+While the conventional approach is to pad variable length sequences, NestedTensor
+enables users to bypass padding. The API for calling operations on a nested tensor is no different
+from that of a regular ``torch.Tensor``, which should allow seamless integration with existing models,
+with the main difference being :ref:`construction of the inputs <construction>`.
+
+As this is a prototype feature, the :ref:`operations supported <supported operations>` are still
+limited. However, we welcome issues, feature requests and contributions. More information on contributing can be found
+`on this wiki <https://github.com/pytorch/pytorch/wiki/NestedTensor-Backend>`_.
```
Comment (mikaylagawarecki) [inline docs/source/nested.rst:27]: note: this wiki has yet to be properly updated

**Review by cpuhrsch — APPROVED** [review]

**Pull request closed without merging**